quinn/src/send_stream.rs

```diff
@@ -122,6 +124,10 @@ impl SendStream {
 
     #[doc(hidden)]
     pub fn poll_finish(&mut self, cx: &mut Context) -> Poll<Result<(), WriteError>> {
+        if self.finished {
```

Comment:
Actually, is there any reason we shouldn't fuse this to `Ok(())`?

Reply:
My preference would be to return `UnknownStream` for consistency rather than tracking additional state. Easy enough for callers to ignore when they really do want to make potentially redundant calls. Also compare `Recv::stop` in proto.